membership-attribute-service/app/components/TouchpointComponents.scala

```diff
@@ -121,18 +122,27 @@ class TouchpointComponents(
     zuoraRestServiceOverride.getOrElse(simpleClientZuoraRestService)
   }
 
-  lazy val catalogRestClient = rest.SimpleClient[Future](backendConfig.zuoraRest, RequestRunners.configurableFutureRunner(60.seconds))
-  lazy val futureCatalog: Future[Catalog] = catalogServiceOverride.getOrElse(
-    CatalogService
-      .read(FetchCatalog.fromZuoraApi(catalogRestClient)(implicitly, LogPrefix.noLogPrefix), productIds)
+  private lazy val futureCatalogNoPrefix: Future[Catalog] = {
+    logger.infoNoPrefix(s"Loading catalog in $stage")
+    val catalogRestClient = rest.SimpleClient[Future](backendConfig.zuoraRest, RequestRunners.configurableFutureRunner(60.seconds))
+    catalogServiceOverride.getOrElse(
+      CatalogService
+        .read(FetchCatalog.fromZuoraApi(catalogRestClient)(implicitly, LogPrefix.noLogPrefix), productIds)
+        .flatMap {
+          case -\/(error) => Future.failed(new RuntimeException(error))
+          case \/-(result) => Future.successful(result)
+        },
+    )
+  }
+  def futureCatalog(implicit logPrefix: LogPrefix): Future[Catalog] =
+    futureCatalogNoPrefix
       .recover { case error =>
-        logger.errorNoPrefix(scrub"Failed to load the product catalog from Zuora due to: $error")
+        logger.error(scrub"Failed to load the product catalog from Zuora due to: $error")
         throw error
```

Comment:
I wasn't sure if this line was supposed to prevent the lazy val from initialising if it fails (and thus retry) but that doesn't seem to be possible.  The catalog fetch is attempted once, and after that if it fails, the server is basically useless.
If it's in PROD there is a healthcheck but it doesn't look like it would fail so it wouldn't be replaced,(the health check intentionally doesn't include test users)

Maybe one for another PR to add catalog fetch to the health check (and/or make it retry)